File: CSharp/Strategy/Ducks/MiniDuckSimualtor/RubberDuck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Strategy
{
    public class RubberDuck : Duck
    {
        public RubberDuck()
        {
            flyBehaviour = new FlyNoWay();
            quackBehaviour = new Squeak();
        }

        public override void display()
        {
            Console.Out.WriteLine("I am a Rubber Duck");
        }
    }
}

File: Java/hfdp/src/main/java/chasej/hfdp/strategy/RubberDuck.java
package chasej.hfdp.strategy;

public class RubberDuck extends Duck {
	//Constructor
	public RubberDuck() {
		flyBehavior = new FlyNoWay();
		quackBehavior = new Squeak();
	}

	public void display() {
		System.out.println("I'm a real Mallard Duck");
	}

}

